docs/source/reference/modding/extended.rst

```diff
@@ -203,8 +203,76 @@ Shield holstering is not supported at the moment since it conflicts with any mod
 
 An example of a mod which uses this feature is `Weapon Sheathing`_.
 
+
+Skeletons extensions
+--------------------
+
+Currently it is possible to inject custom bones to actor's skeleton:
+
+::
+
+    [Game]
+    use additional anim sources = true
+
+If this setting is enabled, OpenMW will seek for modified skeletons in the ``Animations/[skeleton name]`` folder in your ``Data Files``.
+For example, for basic skeleton for biped creatures you need the ``Animations/xbase_anim``, for females you need the ``Animations/xbase_anim_fmnale``, for
+beast races - ``Animations/xbase_anim_kna``. Keep in mind that 1st-person meshes have their own skeletons.
+
+OpenMW scans each NIF file in such folder for nodes with the "BONE" NiStringExtraData.
+It is recommended to give such nodes names starting with "Bip01 " to avoid issues with mesh optimizer.
+Then OpenMW copies all found nodes to related skeleton. To determine to which bone a new node should be attached, OpenMW checks a name of parent node in your NIF file.
+For example, to attach an additional weapon bone, you need such NIF file:
+
+::
+
+NiNode "root"
+    NiNode "Bip01 L Hand"
+        NiNode "Weapon Bone Left"
+            NiStringExtraData "BONE"
+
+In this example OpenMW detects the ``Weapon Bone Left`` node and attaches it to the ``Bip01 L Hand`` node in target skeleton.
+
+An example of a mod which uses this feature is `Weapon Sheathing`_.
+
+
+Extended weapon animations
+--------------------------
+
+It is possible to use unique animation groups for different weapon types. If these animations are absent, fallback ones are used, so new animations are optional.
+Each weapon type has an attack animation ("long" group), and movement animations with given suffix.
+For example, long blades use the ``weapononehand`` animation group for attacks, ``idle1h`` as idle animation, ``jump1h`` as jumping animation, etc.
```

Comment:
For example, long blades use ``weapononehand`` attack animation group, ``idle1h`` idle animation group, ``jump1h`` jumping animation group, etc.